Applications/Spire/Source/Ui/ComboBox.cpp

```diff
@@ -354,6 +355,7 @@ void ComboBox::on_drop_down_submit(const std::any& submission) {
   }
   m_submission = submission;
   m_submission_text = text;
+  QApplication::setActiveWindow(window());
```

Comment:
The [documentation](https://doc.qt.io/qt-5/qapplication.html#setActiveWindow) for this is confusing. It says it doesn't set the keyboard focus on the window, but then below that says it sets the focusWidget (which will receive key input). Is this equivalent to QWidget::activateWindow, or does it actually activate the window without setting keyboard focus?

Reply:
I am also confused about that it doesn't set the keyboard focus and it still sends `FocusIn` event to the widget. The problem here I am trying to solve is to avoid the `OpenFilterPanel` being closed when clicking on the `DropDownList` to choose an item rather than to resolve focus issue. 

Both the `OverlayPanel` of the `OpenFilterPanel` and the `OverlayPanel` of the `DropDownList` inside the `ComboBox` have `Qt::Tool` window flag, and they don't become active by default when they are shown. When clicking on `DropDownList`, the `OverlayPanel` of the `OpenFilterPanel` will gets `FocusObserver::State::NONE` and be closed, which is not supposed to be closed. Calling `QApplication::setActiveWindow` when getting a submit signal from the `DropDownList` can prevent the `OverlayPanel` receiving `FocusObserver::State::NONE`.

Reply:
You might be able to verify this by setting a breakpoint in [setActiveWindow](https://code.woboq.org/qt5/qtbase/src/widgets/kernel/qapplication.cpp.html#2080), but it looks like what the documentation means when it 'sets' the focusWidget is that it actually clears it.

Reply:
I debugged it by setting a breakpoint in 'QApplication::setActiveWindow', For this case, it calls `w->setFocus(Qt::ActiveWindowFocusReason);` on line 2084. 'QApplication::setActiveWindow' does two things: 

1. to deactivate current active window and activate the window passed in; 
2. to set the foucsWidget, which makes the widget in the window passed in get focus.

It does what the documentation says.